### PR TITLE
fix actions in "who has access" table

### LINKED ISSUE #3534: Manage access not working for groups
I just tried to share a resource with a group.  I find that I am able to set group access but not remove or change it.

This is a regression suspected due to vue.js

image

### PR DESCRIPTION
**Pull request by devincowan**
### Pull Request Checklist: 
- [X] Positive Test Case Written by Dev

- [ ] Automated Testing

- [ ] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. Manage who has access on a resource that you own
2. "Give Access" to a different user and a different group. Importantly, the user id must match the group id. To ensure this, I checked the "auth_user" and "auth_user_groups" tables in the db. You could of course python model query or use browser console and USERS_JSON to find a set of group+user whose ids match.
3. After adding the user and group, you will notice that the "undo share" and "remove" actions are still available (whereas before this PR, they were missing).

Resolves #3534

***
BEFORE:
63315526-ccfea900-c2c8-11e9-983a-a4a423038618

Screen Shot 2022-03-03 at 3 05 53 PM

***

AFTER:
Screen Shot 2022-03-03 at 3 08 11 PM

### CODE REVIEW

**Comment by hydrocheck** [comment]
| METRIC | VALUE |
|:-----:|:-----:|

[https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4523](https://sonarqube.cuahsi-workstation.com:9000/dashboard?id=hydroshare-4523)

File: theme/templates/includes/access-row.html [hunk 1]
```diff
@@ -72,15 +72,15 @@
 
     <td class="user-actions">
         <i @click="undoAccess(user, index)"
-           v-if="user.can_undo && quotaHolder.id !== user.id"
+           v-if="user.can_undo && (user.user_type==='group' || (user.user_type === 'user' && quotaHolder.id !== user.id))"
```
Comment (sblack-usu) [inline theme/templates/includes/access-row.html:75]: What is this addressing?  Is there a type that is not user or group?  I can't think of any others, but I also haven't looked up where the user_type is coming from.

File: theme/templates/includes/access-row.html [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {% load hydroshare_tags %}
 
 <tr v-for="(user, index) in this.users" v-bind:key="user.id"
-    :class="{ 'hide-actions': selfAccessLevel !== 'owner' && (user.access === 'owner' && hasOnlyOneOwner || user.user_type === 'user' && user.id === quotaHolder.id), loading: user.loading}">
+    :class="{ 'hide-actions': selfAccessLevel !== 'owner' && (user.access === 'owner' && hasOnlyOneOwner || user.user_type === 'user' && user.id === quotaHolder.id), 'loading': user.loading, 'is-group': user.user_type === 'group'}">
```
Comment (sblack-usu) [inline theme/templates/includes/access-row.html:4]: This is for styling.  Why is there a styling change?

File: theme/templates/includes/access-row.html [hunk 3]
```diff
@@ -72,15 +72,15 @@
...
+           v-if="user.can_undo && (user.user_type==='group' || (user.user_type === 'user' && quotaHolder.id !== user.id))"
             {# TODO: manually initialize tooltips of dynamically created items #}
            class="fa fa-undo" aria-hidden="true" data-toggle="tooltip"
            title="Undo Share"></i>
     </td>
 
     <td class="user-actions">
         <span @click="user.id === currentUser ? showDeleteSelfDialog(user, index) : removeAccess(user, index)"
-              v-if="!(user.access === 'owner' && hasOnlyOneOwner) && user.id !== quotaHolder.id && (selfAccessLevel === 'owner' || user.id === currentUser)"
+              v-if="!(user.access === 'owner' && hasOnlyOneOwner) && !(user.user_type === 'user' && user.id == quotaHolder.id) && (selfAccessLevel === 'owner' || user.id === currentUser)"
```
Comment (sblack-usu) [inline theme/templates/includes/access-row.html:83]: `user.id !== quotaHolder.id`
Why this change?  Why are user id and quotaholder id being compared at all?  Can we remove that?

File: theme/templates/includes/access-row.html (hunk 1 above)
Comment (devincowan) [reply theme/templates/includes/access-row.html:75]: Perhaps I was making assumptions that I shouldn't. You're right that we only have `user_types` of "group" and "user". My thinking is was that we don't want to undo/remove access if the user themself is the quotaHolder. And I was thinking that groups can't be quotaHolders so only checked quotaholder for `user_type === 'user'`

File: theme/templates/includes/access-row.html (hunk 3 above)
Comment (devincowan) [reply theme/templates/includes/access-row.html:83]: same as above, I was assuming that we don't want to remove access for users who are also the quaotHolder for the resource. But I haven't looked into the quota system at all so perhaps this is irrelevant...

File: theme/templates/includes/access-row.html (hunk 2 above)
Comment (sblack-usu) [reply theme/templates/includes/access-row.html:4]: Thanks, sounds good to me.

File: theme/templates/includes/access-row.html (hunk 3 above)
Comment (sblack-usu) [reply theme/templates/includes/access-row.html:83]: Ok, I did not think of why quotaHolder id was important, thanks for pointing that out.  It make sense to me to check quotaHolder.id.

File: theme/templates/includes/access-row.html (hunk 1 above)
Comment (sblack-usu) [reply theme/templates/includes/access-row.html:75]: Makes sense to me with your explanation

**Review by sblack-usu — APPROVED** [review]
